tests/lib/rules/no-misleading-character-class.js

```diff
@@ -272,77 +403,221 @@ ruleTester.run("no-misleading-character-class", rule, {
             code: "var r = /[👨‍👩‍👦]/",
             errors: [
                 {
+                    column: 11,
+                    endColumn: 13,
                     messageId: "surrogatePairWithoutUFlag",
                     suggestions: [{ messageId: "suggestUnicodeFlag", output: "var r = /[👨‍👩‍👦]/u" }]
                 },
                 {
+                    column: 12,
+                    endColumn: 15,
                     messageId: "zwj",
                     suggestions: null
+                },
+                {
+                    column: 14,
+                    endColumn: 16,
+                    messageId: "surrogatePairWithoutUFlag",
+                    suggestions: [{ messageId: "suggestUnicodeFlag", output: "var r = /[👨‍👩‍👦]/u" }]
+                },
+                {
+                    column: 15,
+                    endColumn: 18,
+                    messageId: "zwj"
+                },
+                {
+                    column: 17,
+                    endColumn: 19,
+                    messageId: "surrogatePairWithoutUFlag",
+                    suggestions: [{ messageId: "suggestUnicodeFlag", output: "var r = /[👨‍👩‍👦]/u" }]
                 }
             ]
         },
         {
             code: "var r = /[👨‍👩‍👦]/u",
-            errors: [{
-                messageId: "zwj",
-                suggestions: null
-            }]
+            errors: [
+                {
+                    column: 11,
+                    endColumn: 16,
+                    messageId: "zwj",
+                    suggestions: null
+                },
+                {
+                    column: 14,
+                    endColumn: 19,
+                    messageId: "zwj",
+                    suggestions: null
+                }
+            ]
         },
```

Comment:
This is an interesting case, as `👨‍👩‍👦` appears as a single glyph so it might make the most sense to report only one error that underlines it as a whole (11-19) instead of two overlapping errors (11-16 and 14-19).  Perhaps `zwj()` could assume that a sequence of `<CHAR1><ZWJ><CHAR2><ZWJ><CHAR3>...` represents a grapheme, and return it as a single match?